Source/WebCore/platform/libwpe/PlatformPasteboardLibWPE.cpp

```diff
@@ -84,23 +80,21 @@ String PlatformPasteboard::readString(size_t, const String& type) const
 
 void PlatformPasteboard::write(const PasteboardWebContent& content)
 {
-    static const char plainText[] = "text/plain;charset=utf-8";
-    static const char htmlText[] = "text/html;charset=utf-8";
+    static const auto plainText = "text/plain;charset=utf-8"_s;
+    static const auto htmlText = "text/html;charset=utf-8"_s;
 
     CString textString = content.text.utf8();
     CString markupString = content.markup.utf8();
 
-    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // WPE port
-    struct wpe_pasteboard_string_pair pairs[] = {
+    std::array<struct wpe_pasteboard_string_pair, 2> pairs = { {
         { { nullptr, 0 }, { nullptr, 0 } },
         { { nullptr, 0 }, { nullptr, 0 } },
```

Comment:
Using  `wpe_pasteboard_string_initialize()` copies the strings, then the implementation behind `wpe_pasteboard_write()` will make a second copy, and then each string is freed below with `wpe_pasteboard_string_free()`.

We could initialize the items here with the initializer list, and skip the `wpe_pasteboard_string_free()` call below. Also, notice that there is an `ASCIILiteral::length()` function, we do not need to call `strlen()` manually:

```suggestion
    std::array<struct wpe_pasteboard_string_pair, 2> pairs = { {
        { { plainText, plainText.length() }, { textString.data(), textString.length() } },
        { { htmlText, htmlText.length() }, { markupString.data(), markupString.length() } },
```

Reply:
This doesn't work because `ASCIILiteral` will implicitly convert to a `const char*` and it seems that a `const_cast` to a `char*` is disallowed. For the `CString` part one needs to use `::mutableData()`, but since one can't use the ASCIILiteral directly, I suspect this won't be possible.

Reply:
Fair, I didn't remember that the byte array in `wpe_pasteboard_string` is `char*`, without a `const`. Let's leave it as-is, then.